main.go

```diff
@@ -1691,18 +1691,19 @@ func (tf *transformer) removeUnnecessaryImports(file *ast.File) {
 			return true
 		}
 
-		uses, ok := tf.info.Uses[node].(*types.PkgName)
-		if !ok {
-			return true
+		switch uses := tf.info.Uses[node].(type) {
+		case *types.PkgName:
+			usedImports[uses.Imported().Path()] = true
```

Comment:
I _think_ we don't need this case anymore, because the types.Object case should cover everything.

One can't refer to a package directly, like `foo := importedpkg` - it always has to be via a selector, like `foo := importedpkg.Var` or `foo := importedpkg.Func`. So if we already mark `importedpkg` as used when visiting `Var` and `Func`, we don't need to do it for the package identifier itself.

Reply:
In other words, I don't think we need the type switch - just swap one type assertion with another.

Reply:
Good point, didn't think of that